src/openforms/variables/service.py

```diff
@@ -2,27 +2,62 @@
 Public Python API to access (static) form variables.
 """
 
+from typing import TypedDict
+
 from openforms.forms.models import FormVariable
+from openforms.plugins.registry import BaseRegistry
+from openforms.registrations.registry import register as registration_plugins_registry
 from openforms.submissions.models import Submission
+from openforms.typing import StrOrPromise
 
-from .registry import register_static_variable as register
+from .base import BaseStaticVariable
+from .registry import register_static_variable as static_variables_registry
 
-__all__ = ["get_static_variables"]
+__all__ = ["get_static_variables", "get_registration_plugins_variables"]
 
 
-def get_static_variables(*, submission: Submission | None = None) -> list[FormVariable]:
+def get_static_variables(
+    *,
+    submission: Submission | None = None,
+    variables_registry: BaseRegistry[BaseStaticVariable] | None = None,
+) -> list[FormVariable]:
     """
     Return the full collection of static variables registered by all apps.
 
     :param submission: Optional, but recommended - the submission instance to get the
       variables for. Many of the static variables require the submission for sufficient
       context to be able to produce a value. Variables are static within that submission
       context, i.e. they don't change because of filling out the form.
+    :param variables_registry: The static variables registry to use. If not provided,
+      the default registry will be used.
 
     You should not rely on the order of returned variables, as they are registered in
     the order the Django apps are loaded - and this may change without notice.
     """
     return [
         registered_variable.get_static_variable(submission=submission)
-        for registered_variable in register
+        for registered_variable in variables_registry or static_variables_registry
+    ]
```

Comment:
Can you not provide the `static_variables_registry` as default kwarg instead of `None`?

Otherwise, I prefer the explicit form:

```py
if variables_registry is None:
    variables_registry = static_variables_registry
```
the `or` in the list comprehension is less readable

Reply:
> Can you not provide the `static_variables_registry` as default kwarg instead of `None`?

I went for this in the first place, but felt a bit weird. So I'll go with explicit is None check